src/components/text-editor/text-editor.component.tsx

```diff
@@ -62,41 +98,61 @@ const TextEditor = React.forwardRef(
       previews,
       onLinkAdded,
       ...rest
-    },
+    }: TextEditorProps,
     ref
   ) => {
     const [isFocused, setIsFocused] = useState(false);
-    const [inlines, setInlines] = useState([]);
-    const [activeInlines, setActiveInlines] = useState({});
+    const [inlines, setInlines] = useState<InlineStyleType[]>([]);
+    const [activeInlines, setActiveInlines] = useState<{
+      BOLD?: boolean;
+      ITALIC?: boolean;
+    }>({});
     const [focusToolbar, setFocusToolbar] = useState(false);
 
-    const editorRef = useRef();
+    const editorRef = useRef<Editor>(null);
     const editor = ref || editorRef;
     const contentLength = getContent(value).getPlainText("").length;
     const moveCursor = useRef(contentLength > 0);
-    const lastKeyPressed = useRef();
+    const lastKeyPressed = useRef<null | string>();
     const labelId = useRef(`text-editor-label-${createGuid()}`);
 
-    const keyBindingFn = (ev) => {
+    const invalidRowProp = rows && (typeof rows !== "number" || rows < 2);
+    invariant(
+      !invalidRowProp,
+      `Prop rows must be a number value greater than 2 to override the min-height of the \`${TextEditor.displayName}\``
+    );
+
+    const keyBindingFn = (ev: React.KeyboardEvent<HTMLInputElement>) => {
       if (Events.isTabKey(ev) && !Events.isShiftKey(ev)) {
         setFocusToolbar(true);
       }
 
       return getDefaultKeyBinding(ev);
     };
 
-    const handleKeyCommand = (command) => {
+    const isBlock = (str?: string) => {
+      const { blockType } = getContentInfo(value);
+      const newBlockType = str ? computeBlockType(str, blockType) : null;
+
+      if (ORDERED_LIST === blockType || UNORDERED_LIST === blockType)
+        return true;
+      if (ORDERED_LIST === newBlockType || UNORDERED_LIST === newBlockType)
+        return true;
+      return false;
+    };
+
+    const handleKeyCommand = (command: EditorCommand) => {
       // bail out if the enter is pressed and limit has been reached
       if (command.includes("split-block") && contentLength === characterLimit) {
         return "handled";
       }
 
       // if the backspace or enter is pressed get block type and text
       if (command.includes("backspace") || command.includes("split-block")) {
-        const { blockType, blockLength } = getContentInfo(value);
+        const { blockLength } = getContentInfo(value);
 
         // if a block control is active and there is no text, deactivate it and reset the block
-        if (BLOCK_TYPES.includes(blockType) && !blockLength) {
+        if (isBlock() && !blockLength) {
```

Comment:
I'm unfamiliar with the intended logic here, but it looks to me like this change will update the behaviour. Previously this would only come up `true` if `blockType` was one of the two strings we're checking against, but now we also allow the `newBlockType` returned from `computeBlockType` to be one of these values. I don't know what difference this makes in practice, but assuming it does it's worth looking at if this wasn't an intentional change. (And if they end up giving the same answer I prefer the old version as it was simpler to understand!)

Similar comments apply to the other place where `isBlock` is called.

Reply:
I'll reverted these changes back to what they were. We are confident that way worked so we'll stick with that for now.